src/test/java/org/wildfly/security/sasl/oauth2/OAuth2SaslClientTest.java

```diff
@@ -158,8 +157,8 @@ public void failedWithBearerTokenFromConfiguration() throws Exception {
         try {
             message = saslServer.evaluateResponse(message);
         } catch (SaslException e) {
-            assertTrue(e.getCause() instanceof RealmUnavailableException);
-            RealmUnavailableException cause = (RealmUnavailableException) e.getCause();
+            assertTrue(e.getCause() instanceof IllegalArgumentException);
```

Comment:
It is just a test method asserting a specific message during authentication. In real world code we are not supposed to rely on any other message but ```SaslException```.